src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/reports/sessiondays/SessionDaysReport.java

```diff
@@ -0,0 +1,271 @@
+                        }
+                        reportDetail.setCaseReference(caseData.getEthosCaseReference());
+                        reportDetail.setHearingNumber(hearingTypeItem.getValue().getHearingNumber());
+                        reportDetail.setHearingType(hearingTypeItem.getValue().getHearingType());
+                        reportDetail.setHearingSitAlone("Sit Alone".equals(
+                                hearingTypeItem.getValue().getHearingSitAlone()) ? "Y" : "");
+                        setTelCon(hearingTypeItem, reportDetail);
+                        String duration = calculateDuration(dateListedTypeItem);
+                        reportDetail.setHearingDuration(duration);
+                        reportDetail.setSessionType(getSessionType(duration));
+                        reportDetail.setHearingClerk(dateListedTypeItem.getValue().getHearingClerk());
+                        reportDetailList.add(reportDetail);
+                    }
+                }
+            }
+        }
+    }
+
+    private void setTelCon(HearingTypeItem hearingTypeItem, SessionDaysReportDetail reportDetail) {
+        if (HEARING_TYPE_JUDICIAL_MEDIATION_TCC.equals(hearingTypeItem.getValue().getHearingType())) {
+            reportDetail.setHearingTelConf("Y");
+        } else {
+            reportDetail.setHearingTelConf("");
+        }
+    }
+
+    private String calculateDuration(DateListedTypeItem c) {
+        var dateListedType = c.getValue();
+        long duration = 0;
+        long breakDuration = 0;
+
+        var hearingTimingBreak = dateListedType.getHearingTimingBreak();
+        var hearingTimingResume = dateListedType.getHearingTimingResume();
+        //If there was a break and resumption during the hearing
+        if (!StringUtil.isNullOrEmpty(hearingTimingBreak)
+                && !StringUtil.isNullOrEmpty(hearingTimingResume)) {
+            var hearingBreak = LocalDateTime.parse(hearingTimingBreak, OLD_DATE_TIME_PATTERN);
+            var hearingResume = LocalDateTime.parse(hearingTimingResume, OLD_DATE_TIME_PATTERN);
+            breakDuration = ChronoUnit.MINUTES.between(hearingBreak, hearingResume);
+        }
+
+        var hearingTimingStart = dateListedType.getHearingTimingStart();
+        var hearingTimingFinish = dateListedType.getHearingTimingFinish();
+        if (!StringUtil.isNullOrEmpty(hearingTimingStart)
+                && !StringUtil.isNullOrEmpty(hearingTimingFinish)) {
+            var hearingStartTime = LocalDateTime.parse(hearingTimingStart, OLD_DATE_TIME_PATTERN);
+            var hearingEndTime = LocalDateTime.parse(hearingTimingFinish, OLD_DATE_TIME_PATTERN);
+            long startToEndDiffInMinutes = ChronoUnit.MINUTES.between(hearingStartTime, hearingEndTime);
+            duration = startToEndDiffInMinutes - breakDuration;
+        }
+
+        return String.valueOf(duration);
+    }
+
+    private String getSessionType(String hearingDuration) {
+        String sessionType = NONE;
```

Comment:
You don't need this really. Just return the value directly

Reply:
refactored